pkg/controllers/management/auth/globalroles/globalrolebinding_handler.go

```diff
@@ -585,6 +598,127 @@ func (grb *globalRoleBindingLifecycle) grantRestrictedAdminUserClusterPermission
 	return returnErr
 }
 
+// reconcileNamespacedRoleBindings ensures that RoleBindings exist for each namespace listed in NamespacedRules
+// from the associated GlobalRole
+func (grb *globalRoleBindingLifecycle) reconcileNamespacedRoleBindings(globalRoleBinding *v3.GlobalRoleBinding) error {
+	var returnError error
+	grbName := wrangler.SafeConcatName(globalRoleBinding.Name)
+	gr, err := grb.grLister.Get("", globalRoleBinding.GlobalRoleName)
+	if err != nil {
+		return fmt.Errorf("unable to get globalRole %s: %w", globalRoleBinding.GlobalRoleName, err)
+	}
+
+	roleBindingUIDs := map[types.UID]struct{}{}
+
+	for ns := range gr.NamespacedRules {
+		namespace, err := grb.nsLister.Get("", ns)
+		if apierrors.IsNotFound(err) {
```

Comment:
Might also be good to check if the returned namespace was nil

```suggestion
		if apierrors.IsNotFound(err) || namespace == nil {
```